modules/Microsoft.Storage/storageAccounts/blobServices/deploy.bicep

```diff
@@ -57,7 +54,7 @@ param diagnosticMetricsToEnable array = [
 ]
 
 @description('Optional. The name of the diagnostic setting, if deployed.')
-param diagnosticSettingsName string = '${name}-diagnosticSettings'
+param diagnosticSettingsName string = 'default-diagnosticSettings'
```

Comment:
I'd keep the optional diagnostic settings feature as aligned as possible across all modules. In this and other cases below I'd suggest using a variable `name` since the hardcoded value is used in more than one place

Reply:
hm ok

Reply:
Just tried and actually - no. We cannot reference variables in parameters.

Reply:
Just opened a draft PR #2814 with what I meant

Reply:
Looks good. The only 'caviat' is, that now the user is actually not aware of this default value as it is not published to the readme.

Reply:
True, in that case we should probably add something in the description, e.g., "defaults to ... if empty"? In case this alternative makes sense, I guess it should be implemented in a consistent way also for modules not having an hardcoded name? Maybe worth a quick internal discussion.

Reply:
Following a discussion we'll go with the suggestion by @eriqua to keep the diagnostic settings aligned & reduce the amount of duplicated code/values

Reply:
We can resolve this comment once the PR was merged into this